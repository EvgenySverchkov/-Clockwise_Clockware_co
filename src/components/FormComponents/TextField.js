import React, {useContext} from "react";
import PropTypes from "prop-types";

import Tooltip from "../CustomTooltip";
import Context from "../../ContextComponent";

const TextField = ({ id, name, chngHandler, value }) => {
  const context = useContext(Context);
  return (
  <div className="col-sm-8 col-md-8">
    <Tooltip 
      title={context.warningTooltipMsg} 
      open={context.isOpenWarningTooltip} 
      fieldId = {context.idForTooltip}
      tooltipIsOpen = {context.isOpenWarningTooltip}>
    <input
      id={id}
      type="text"
      name={name || null}
      className="form-control"
      onChange={chngHandler || null}
      value={value}
      required
    />
    </Tooltip>
  </div>);
}

TextField.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string,
  chngHandler: PropTypes.func,
  value: PropTypes.node,
};

export default TextField;
