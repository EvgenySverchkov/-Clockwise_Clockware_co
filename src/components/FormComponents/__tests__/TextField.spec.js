import React from 'react';
import {create} from "react-test-renderer";

import TextField from "../TextField";
import ContextComponent from "../../../ContextComponent";

it("<TextField /> test which is used in forms", ()=>{
    const mockProps = {
        id: "1", 
        name: "Name", 
        chngHandler: jest.fn(), 
        value: "Value"
    }
    const mockContextValue = {
        warningTooltipMsg: "Message",
        isOpenWarningTooltip: false,
        idForTooltip: 1
    }
    const component = create(
        <ContextComponent.Provider value={mockContextValue}>
            <TextField {...mockProps}/>
        </ContextComponent.Provider>
    )
    expect(component.toJSON()).toMatchSnapshot();
});