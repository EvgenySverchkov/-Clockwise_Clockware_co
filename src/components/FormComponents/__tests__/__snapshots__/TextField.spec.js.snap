// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> test which is used in forms 1`] = `
<div
  className="col-sm-8 col-md-8"
>
  <input
    aria-describedby={null}
    className="form-control"
    id="1"
    name="Name"
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    required={true}
    title="Message"
    type="text"
    value="Value"
  />
</div>
`;
